Typing error when sending to parent

## Bug

## Description:

When using `sendParent` in an actions array with the function version, TypeScirpt complains about the `states` property of the machine not being valid. Setting the types of the parent yield the same result: `sendParent<ParentContext, ParentEvent>(...)`

## (Bug) Expected result:

No TS error

## (Bug) Actual result:

```
yarn run v1.17.3
$ /Users/posva/xstate-ts-bug/node_modules/.bin/tsc
src/index.ts(4,3): error TS2322: Type '{ one: { on: { '': { target: string; actions: (SendAction<unknown, EventObject> | SendAction<unknown, { type: "parent"; ctx: unknown; }>)[]; }; }; }; two: {}; }' is not assignable to type 'StatesConfig<DefaultContext, any, EventObject>'.
  Property 'one' is incompatible with index signature.
    Type '{ on: { '': { target: string; actions: (SendAction<unknown, EventObject> | SendAction<unknown, { type: "parent"; ctx: unknown; }>)[]; }; }; }' is not assignable to type 'StateNodeConfig<DefaultContext, any, EventObject>'.
      Types of property 'on' are incompatible.
        Type '{ '': { target: string; actions: (SendAction<unknown, EventObject> | SendAction<unknown, { type: "parent"; ctx: unknown; }>)[]; }; }' is not assignable to type 'TransitionsConfig<DefaultContext, EventObject>'.
          Property '''' is incompatible with index signature.
            Type '{ target: string; actions: (SendAction<unknown, EventObject> | SendAction<unknown, { type: "parent"; ctx: unknown; }>)[]; }' is not assignable to type 'string | number | StateNode<DefaultContext, any, EventObject> | TransitionConfig<DefaultContext, EventObject> | TransitionConfig<DefaultContext, EventObject>[] | undefined'.
              Type '{ target: string; actions: (SendAction<unknown, EventObject> | SendAction<unknown, { type: "parent"; ctx: unknown; }>)[]; }' is not assignable to type 'undefined'.
```

## (Bug) Potential fix:

I think the type should be a bit looser as using a simple `sendParent('parent')` doesn't yield any error


## Link to reproduction or proof-of-concept:

I'm sorry I couldn't reproduce the TS error on a codesandbox but this is a fully boiled down repro with only typescript and xstate as deps

https://github.com/posva/xstate-ts-bug

repro steps:

- yarn
- yarn tsc

--- a/.changeset/tame-turkeys-live.md
+++ b/.changeset/tame-turkeys-live.md
@@ -1,0 +1,5 @@
+---
+'xstate': patch
+---
+
+Loosened event type for `SendAction<TContext, AnyEventObject>`

--- a/packages/core/src/types.ts
+++ b/packages/core/src/types.ts
@@ -70,7 +70,7 @@ export type Action<TContext, TEvent extends EventObject> =
   | ActionObject<TContext, TEvent>
   | ActionFunction<TContext, TEvent>
   | AssignAction<Required<TContext>, TEvent>
-  | SendAction<TContext, TEvent>
+  | SendAction<TContext, AnyEventObject>
   | RaiseAction<AnyEventObject>;
 
 export type Actions<TContext, TEvent extends EventObject> = SingleOrArray<

